src/github/githubRepository.ts

```diff
@@ -203,103 +203,44 @@ export class GitHubRepository implements IGitHubRepository, vscode.Disposable {
 		};
 	}
 
-	async getPullRequests(prType: PRType, page?: number): Promise<PullRequestData | undefined> {
-		return prType === PRType.All ? this.getAllPullRequests(page) : this.getPullRequestsForCategory(prType, page);
-	}
+	async getPullRequestsGraphQL(type: PRType, nextCursor?: string|null):Promise<PullRequestListResponse|undefined> {
+		const { remote, query, octokit } = await this.ensure();
+		const currentUser = octokit && (octokit as any).currentUser;
+		const currentUserLogin: string = currentUser.login;
 
-	private async getAllPullRequests(page?: number): Promise<PullRequestData | undefined> {
-		try {
-			Logger.debug(`Fetch all pull requests - enter`, GitHubRepository.ID);
-			const { octokit, remote } = await this.ensure();
-			const result = await octokit.pullRequests.getAll({
-				owner: remote.owner,
-				repo: remote.repositoryName,
-				per_page: PULL_REQUEST_PAGE_SIZE,
-				page: page || 1
-			});
-
-			const hasMorePages = !!result.headers.link && result.headers.link.indexOf('rel="next"') > -1;
-			const pullRequests = result.data
-				.map(
-					pullRequest => {
-						if (!pullRequest.head.repo) {
-							Logger.appendLine(
-								'GitHubRepository> The remote branch for this PR was already deleted.'
-							);
-							return null;
-						}
-
-						const item = convertRESTPullRequestToRawPullRequest(pullRequest);
-						return new PullRequestModel(this, this.remote, item);
-					}
-				)
-				.filter(item => item !== null) as PullRequestModel[];
+		let filter = `type:pr is:open repo:${remote.owner}/${remote.repositoryName}`;
 
-			Logger.debug(`Fetch all pull requests - done`, GitHubRepository.ID);
-			return {
-				pullRequests,
-				hasMorePages
-			};
-		} catch (e) {
-			Logger.appendLine(`Fetching all pull requests failed: ${e}`, GitHubRepository.ID);
-			if (e.code === 404) {
-				// not found
-				vscode.window.showWarningMessage(`Fetching pull requests for remote '${this.remote.remoteName}' failed, please check if the url ${this.remote.url} is valid.`);
+		if (type !== PRType.All) {
+			if (type === PRType.Mine) {
+				filter += ` author:${currentUserLogin}`;
+			} else if (type === PRType.RequestReview) {
+				filter += ` review-requested:${currentUserLogin}`;
+			} else if (type === PRType.AssignedToMe) {
+				filter += ` assignee:${currentUserLogin}`;
 			} else {
-				throw e;
+				throw new Error('Unexpected pull request filter');

```

Comment:
would it make sense to log the filter type here?

Reply:
:+1: